twitter_scrooge/BUILD

```diff
@@ -0,0 +1,83 @@
+load("//twitter_scrooge/toolchain:toolchain.bzl", "export_scrooge_deps", "scrooge_toolchain")
+load("//scala:providers.bzl", "declare_deps_provider")
+
+scrooge_toolchain(
+    name = "scrooge_toolchain_impl",
+    visibility = ["//visibility:public"],
+)
+
+toolchain(
+    name = "scrooge_toolchain",
+    toolchain = ":scrooge_toolchain_impl",
+    toolchain_type = "@io_bazel_rules_scala//twitter_scrooge/toolchain:scrooge_toolchain_type",
+    visibility = ["//visibility:public"],
+)
+
+declare_deps_provider(
+    name = "aspect_compile_classpath_provider",
+    deps_id = "aspect_compile_classpath",
+    visibility = ["//visibility:public"],
+    deps = [
+        "//external:io_bazel_rules_scala/dependency/thrift/javax_annotation_api",
+        "//external:io_bazel_rules_scala/dependency/thrift/libthrift",
+        "//external:io_bazel_rules_scala/dependency/thrift/scrooge_core",
+        "//external:io_bazel_rules_scala/dependency/thrift/util_core",
+        "//scala/private/toolchain_deps:scala_library_classpath",
+    ],
+)
+
+declare_deps_provider(
+    name = "compile_classpath_provider",
+    deps_id = "compile_classpath",
+    visibility = ["//visibility:public"],
+    deps = [
+        "//external:io_bazel_rules_scala/dependency/thrift/libthrift",
+        "//external:io_bazel_rules_scala/dependency/thrift/scrooge_core",
+        "//scala/private/toolchain_deps:scala_library_classpath",
+    ],
+)
+
+declare_deps_provider(
+    name = "scrooge_generator_provider",
```

Comment:
Given that the other groupings are named "by intention", could we name this something like `scrooge_worker_classpath_provider`?

I feel like the goal here is to allow someone to grow and shrink these classpaths separately, so even if someone wants to implement a rule that uses "just the scrooge generator", they still shouldn't to use this dep provider.

Reply:
done